stdlib/path.gr

```diff
@@ -0,0 +1,774 @@
+/**
+ * @module Path: Utilities for working with system paths.
+ * 
+ * This module treats paths purely as a data representation and does not
+ * provide functionality for interacting with the file system.
+ * 
+ * This module explicitly encodes whether a path is absolute or relative, and
+ * whether it refers to a file or a directory, as part of the `Path` type.
+ *
+ * Paths in this module abide by a special POSIX-like representation/grammar
+ * rather than one defined by a specific operating system. The rules are as
+ * follows:
+ * 
+ * - Path separators are denoted by `/` for POSIX-like paths and additionally `\` for Windows paths
```

Comment:
```suggestion
 * - Path separators are denoted by `/` for POSIX-like paths
```
I don't think we need to mention Windows paths here as long as this distinction is mentioned on `fromPlatformString` and `toPlatformString`